### ISSUE TITLE
HOWTO Recognize 4x4 markers

### ISSUE THREAD

**Issue by mrcesc0** [artoolkitx/jsartoolkit5#109]
Hello,
I'd like to build a react native app able to read **4x4 markers**, so a number of 8191 markers.
Which way should i follow to achieve that?

I took a look at this example https://github.com/artoolkitx/jsartoolkit5/blob/master/examples/multimarker_barcode_threejs.html but have some doubts on how to generate a *.dat file (as in the example: **arController.loadMultiMarker('Data/multi-barcode-4x3.dat'**) that contains all the markers I need and if that's the proper way to the reach the goal or i should consider another method.

Thank you so much
Francesco

**Comment by ThorstenBux** [comment]
Hi, what you want is basically these pattern markers: https://github.com/artoolkit/ARToolKit5/tree/master/doc/patterns
which are already preconfigured inside artoolkit so you don't need to create the patt files. However, the 4x4 markers are not really documented only the 3x3. Have a look at the artoolkit5 examples how the 3x3 are loaded. that should give you an idea how you can use them. And maybe give a clue how to use the 4x4. They may be in a Unity3D plugin somewhere, I'm not sure.

**Comment by mrcesc0** [comment]
Thank you for your answer. 
Since you wrote "which are already preconfigured inside artoolkit", I'm wondering if there is a way to recognize a marker's code without load/declare them all.

**Comment by ThorstenBux** [comment]
They are pre-configured in a way that you don't need to create the patt files. You still need to tell the tracker which markers you are looking for. Meaning you need to 'load' (I guess instantiate is the better word) them.

**Comment by mrcesc0** [comment]
Everything's clear. That's the answer i was looking for...
Anyway I'm a bit concerned about the performance issues for instantiating 8191 markers.

Thank  you!!!

**Comment by ThorstenBux** [comment]
Yeah, that I can't say. And honestly, I'm wondering why you would want that many markers at once.

**Comment by mrcesc0** [comment]
Every marker would be associated to a specific customer, so I need them all in order to have as many "available slots" as possible.

**Comment by ThorstenBux** [comment]
Hm, I guess I would need to understand your complete app to get what you are aiming for :).

**Comment by kalwalt** [comment]
Hi @mrcesc0 the multi marker feature Is intended to be used ti improve the detection and tracking, in the sense more markers acting like only one.  I can point you to the docs, i have a repo in my profile, but searching on my smartphone Is not easy...🙂
